tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -926,4 +939,89 @@ public function renameGroup(string $oldGroup, string $newGroup, ?string $user =
 	public function userRenamesGroupUsingTheGraphApi(string $user, string $oldGroup, string $newGroup): void {
 		$this->featureContext->setResponse($this->renameGroup($oldGroup, $newGroup, $user));
 	}
+
+	/**
+	 * @param string $user
+	 *
+	 * @return ResponseInterface
+	 * @throws JsonException
+	 * @throws GuzzleException
+	 */
+	public function retrieveUserInformationUsingGraphApi(
+		string $user
+	):ResponseInterface {
+		$credentials = $this->getAdminOrUserCredentials($user);
+		return GraphHelper::getOwnInformationWithGroups(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$credentials["username"],
+			$credentials["password"],
+		);
+	}
+
+	/**
+	 * @When /^the user "([^"]*)" retrieves (her|his) information using the Graph API$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws JsonException
+	 */
+	public function userRetrievesHisorHerInformationOfUserUsingGraphApi(
+		string $user
+	):void {
+		$response = $this->retrieveUserInformationUsingGraphApi($user);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @Then /^the API response should contain the following information:$/
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function theUserRetrieveApiResponseShouldContainTheFollowingInformation(TableNode $table): void {
+		$rows = $table->getRowsHash();
+		$apiResponse = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse());
+
+		foreach (array_keys($rows) as $keyName) {
+			if ($keyName === 'memberOf') {
+				$memberOfFromApiReponse = [];
+				$memberOf = preg_split('/\s*,\s*/', trim($rows['memberOf']));
+				foreach ($apiResponse['memberOf'] as $member) {
+					$memberOfFromApiReponse[] = $member['displayName'];
+				}
+				Assert::assertEqualsCanonicalizing($memberOf, $memberOfFromApiReponse);
+			} elseif ($keyName === 'id') {
+				// check user_id
+				$isUUIDv4 = $this->featureContext->substituteInLineCodes(
+					$rows['id'],
+					$this->featureContext->getCurrentUser(),
+					[],
+					[
+						[
+							"code" => "%UUIDv4%",
+							"function" =>
+								[$this, "checkUUIDv4PatternForUserId"],
+							"parameter" => [$apiResponse['id']]
+						],
+					]
+				);
```

Comment:
why using substitution here? what u actually substituting? I think u just assert the ID in the response has proper UUID func. which can be done easily just by calling that `checkUUIDv4...` func.
```
Assert::assertTrue($this->checkUUID....($apiResponse['id']);
```